jsonrpc/src/main/scala/org/langmeta/jsonrpc/MessageWriter.scala

```diff
@@ -20,7 +20,7 @@ import io.circe.syntax._
  *
  * @note The header part is defined to be ASCII encoded, while the content part is UTF8.
  */
-class MessageWriter(out: OutputStream) extends LazyLogging {
+class MessageWriter(out: OutputStream, logger: Logger) {
```

Comment:
Note that this changes the logger name: `LazyLogging` instantiates a logger using `getClass.getName` as the name, so now this logger won't be called `org.langmeta.jsonrpc.MessageWriter`, but something else depending on the caller.

If we move to file/position kind of logging instead this is won't be an issue, but for now this will lead to inconsistent names in the logs.

A possible alternative is to explicitly do

```scala
val logger = Logger(classOf[MessageWriter])
```

on the call site. It's not ideal, but at least it's consistent.

Reply:
I think that
```scala
val logger = Logger(classOf[MessageWriter])
```
will use LoggerFactory which breaks the integration in bloop. I agree this change makes debugging quite bad, but I personally use `pprint` during development. I would love to switch to a file:line output format where this won't matter, but that would mean moving to another api like airframe-log (which is not sl4j backed). What do you propose we do?

Reply:
How does `LoggerFactory` break the integration in bloop? `LazyLogging` uses `LoggerFactory` too:

 https://github.com/typesafehub/scala-logging/blob/master/src/main/scala/com/typesafe/scalalogging/Logging.scala

Reply:
bloop needs a way to use the jsonrpc API without triggering LoggerFactory, LazyLogging is behind a lazy val so if it's never used then LoggerFactory never gets triggered.